packages/core/src/components/MultiSelect/Options/Options.tsx

```diff
@@ -26,6 +37,13 @@ const Component: FC<OptionsProps> = memo(props => {
             },
             [selectedValues, onOptionClick]
         ),
+        handleSelectionFromKeyboard = useCallback(
```

Comment:
Can we extract out the common logic from `handleCheckboxClick` & `handleSelectionFromKeyboard`.